primitives/inherents/src/lib.rs

```diff
@@ -433,9 +435,23 @@ pub trait ProvideInherent {
 	/// When checking an inherent, the first parameter represents the inherent that is actually
 	/// included in the block by its author. Whereas the second parameter represents the inherent
 	/// data that the verifying node calculates.
+	///
+	/// NOTE: A block can contains multiple inherent.
 	fn check_inherent(_: &Self::Call, _: &InherentData) -> Result<(), Self::Error> {
 		Ok(())
 	}
+
+	/// Return whereas the call is an inherent call.
+	///
+	/// NOTE: Signed extrinsics are not inherent, but signed extrinsic with the given call variant
+	/// can be dispatched.
+	///
+	/// # Warning
+	///
+	/// Pallets with unsigned transactions **must ensure** that no unsigned transaction call
```

Comment:
Can you elaborate a bit more how someone can shoot themselves in the foot here?

Reply:
I think this trait should be moved to frame, but I tried to give more detail